### PR TITLE
LDAP Login

### LINKED ISSUE #35: Implement LDAP for user login
Connect the login page to verify credentials with the LDAP server rather than a database query.

- [x] Validate LDAP user
- [x] Validate Non-LDAP super user
- [x] Create employee for new LDAP user

### PR DESCRIPTION
**Pull request by blackjack26**
- Uses LDAP credentials to log into system
- Added comment headers in files changed
- Moved Role class to new file
- Moved `./core/database/db_config.ini` to `./config.ini`
    - Added LDAP configuration
    - Updated [wiki config](https://github.com/Capping-CPCA/Web-App/wiki/Server-Set-Up#add-database-user-password)
- Added create account page
- Super user login

Fixes #35

### CODE REVIEW

**Comment by Dr-Crow** [comment]
@blackjack26 I am trying to trace through your code, but front end is not my specialty. Does this pull the email from LDAP and compare it to the database? Also for the username, are you grabbing first name and last name from the database to display on the page?

**Comment by blackjack26** [comment]
Currently it verifies the user through LDAP binding, and if the user is authenticated, I make the email based on their username + @thecpca.com. I couldn't find the email attribute in LDAP so I used their username. Is that how the email is specified @Dr-Crow?

**Comment by vmjoseph** [comment]
Looks good to me, but one question:
Since we're not on their network yet, is there a way to test LDAP to ensure it can connect to there server/ interface with our db?

**Comment by blackjack26** [comment]
I mean the only way to check would be to change the configuration in the `config.ini` file that will contain the LDAP setttings

File: views/create_account.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                die();
+            } else {
+                pg_free_result($res);
+                $error = $state;
+            }
+        } else {
+            $error = $state;
+        }
+    } else {
+        $error = "-1";
```
Comment (Hoenn) [inline views/create_account.php]: This error state should probably be more specific. It doesn't look like it should ever happen, all the more reason to self document.

Comment (ghost) [inline views/create_account.php]: This error should probably have a more specific message. It doesn't look like it _should_ ever happen, all the more reason to self document.

**Review by ghost — APPROVED** [review]
LGTM

**Comment by vmjoseph** [comment]
Everything looks good; love the error handling instead of just detecting.

**Pull request merged**